### PR TITLE
CRUD Auth and Users with HTTP

### PR DESCRIPTION
**Pull request by MartinUgarte**

### CODE REVIEW

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
Comment (ianshih2003) [inline src/modules/auth/auth.service.ts:1]: sacar

File: src/modules/auth/auth.module.ts [hunk 2]
```diff
@@ -1,32 +1,22 @@
...
 import { UserModule } from '../user/user.module';
 import { UserService } from '../user/user.service';
 import { AuthController } from './auth.controller';
 import { AuthService } from './auth.service';
-import { AUTH_PACKAGE_NAME, AUTH_SERVICE_NAME } from './interfaces/auth.pb';
 
 @Module({
   imports: [
+    ConfigModule.forRoot(),
+    HttpModule,
```
Comment (ianshih2003) [inline src/modules/auth/auth.module.ts:14]: para que

**Review by ianshih2003 — CHANGES_REQUESTED** [review]

File: src/modules/auth/auth.service.ts [hunk 3]
```diff
@@ -30,14 +26,13 @@ export class AuthService {
 
   validateNewUser(user: RegisterRequest): void {
     if (!Object.values(Role).includes(RoleTransformer(user.role))) {
-      throw new InvalidArgumentException('Invalid Role');
+      throw new BadRequestException({ message: 'Invalid role' });
```
Comment (ManuSanchez02) [inline src/modules/auth/auth.service.ts:29]: Usa las excepciones correspondientes (https://docs.nestjs.com/exception-filters#built-in-http-exceptions)

File: src/modules/auth/auth.service.ts [hunk 4]
```diff
@@ -113,7 +110,9 @@ export class AuthService {
 
       return this.userService.getUserByEmail(payload.email!);
     } catch (error) {
-      throw new UnauthorizedException('The token is invalid');
+      throw new BadRequestException({
```
Comment (ManuSanchez02) [inline src/modules/auth/auth.service.ts:113]: Aca tmb

File: src/modules/auth/auth.service.ts [hunk 5]
```diff
@@ -59,11 +54,13 @@ export class AuthService {
     } catch (error: any) {
       switch (error.code) {
         case 'auth/email-already-in-use':
-          throw new AlreadyExistsException('Email already in use');
+          throw new BadRequestException({
+            message: `Email already in use: ${error}`,
+          });
         case 'auth/invalid-email':
-          throw new InvalidArgumentException('Invalid email');
+          throw new BadRequestException({ message: `Invalid email: ${error}` });
```
Comment (ManuSanchez02) [inline src/modules/auth/auth.service.ts:61]: Aca tmb

File: src/modules/auth/auth.service.ts [hunk 6]
```diff
@@ -59,11 +54,13 @@ export class AuthService {
...
         case 'auth/weak-password':
-          throw new InvalidArgumentException('Weak password');
+          throw new BadRequestException({ message: `Weak password: ${error}` });
```
Comment (ManuSanchez02) [inline src/modules/auth/auth.service.ts:63]: Aca tmb

File: src/modules/auth/auth.service.ts [hunk 7]
```diff
@@ -59,11 +54,13 @@ export class AuthService {
     } catch (error: any) {
       switch (error.code) {
         case 'auth/email-already-in-use':
-          throw new AlreadyExistsException('Email already in use');
+          throw new BadRequestException({
```
Comment (ManuSanchez02) [inline src/modules/auth/auth.service.ts:57]: Aca tmb

File: src/modules/user/user.controller.ts [hunk 8]
```diff
@@ -1,50 +1,63 @@
...
-    const user = await this.userService.getUserById(data?.id);
+  @Get(':id')
+  async getUserById(
+    @Param('id', ParseIntPipe) id: number
+  ): Promise<User | null> {
+    const user = await this.userService.getUserById(id);
 
     if (!user) {
-      throw new NotFoundException('User not found');
+      throw new BadRequestException({ message: 'User not found' });
```
Comment (ManuSanchez02) [inline src/modules/user/user.controller.ts:27]: Aca tmb

File: src/modules/user/user.controller.ts [hunk 9]
```diff
@@ -1,50 +1,63 @@
...
+  @Put(':id')
+  async putEditUser(
+    @Param('id', ParseIntPipe) id: number,
+    @Body() user: UserDTO
+  ): Promise<User> {
+    const editedUser = await this.userService.editUser(id, user);
 
     if (!editedUser) {
-      throw new NotFoundException('User not found');
+      throw new BadRequestException({ message: 'User not found' });
```
Comment (ManuSanchez02) [inline src/modules/user/user.controller.ts:46]: Aca tmb

File: src/modules/user/user.controller.ts [hunk 10]
```diff
@@ -1,50 +1,63 @@
...
+  async deleteUser(
+    @Param('id', ParseIntPipe) id: number
+  ): Promise<User | undefined> {
     try {
-      return await this.userService.deleteUser(data?.id);
+      return await this.userService.deleteUser(id);
     } catch (e) {
       if ((e as any)?.code === 'P2025') {
-        throw new NotFoundException('User not found');
+        throw new BadRequestException({ message: 'User not found' });
```
Comment (ManuSanchez02) [inline src/modules/user/user.controller.ts:60]: Aca tmb

**Review by ManuSanchez02 — CHANGES_REQUESTED** [review]

**Review by ianshih2003 — CHANGES_REQUESTED** [review]

**Review by ianshih2003 — APPROVED** [review]

**Review by ManuSanchez02 — APPROVED** [review]

**Pull request merged**